app/test/unit/git/cherry-pick-test.ts

```diff
@@ -0,0 +1,250 @@
+import { GitProcess } from 'dugite'
+import * as FSE from 'fs-extra'
+import * as Path from 'path'
+import { getCommit, getCommits, merge, MergeResult } from '../../../src/lib/git'
+import { cherryPick, CherryPickResult } from '../../../src/lib/git/cherry-pick'
+import { Branch } from '../../../src/models/branch'
+import { Repository } from '../../../src/models/repository'
+import { getBranchOrError } from '../../helpers/git'
+import { createRepository } from '../../helpers/repository-builder-cherry-pick-test'
+import {
+  createBranch,
+  makeCommit,
+  switchTo,
+} from '../../helpers/repository-scaffolding'
+
+const featureBranchName = 'this-is-a-feature'
+const targetBranchName = 'target-branch'
+
+describe('git/cherry-pick', () => {
+  let repository: Repository
+  let featureBranch: Branch
+  let targetBranch: Branch
+  let result: CherryPickResult | null
+
+  beforeEach(async () => {
+    // This will create a repository with a feature branch with one commit to
+    // cherry pick and will check out the target branch.
+    repository = await createRepository(featureBranchName, targetBranchName)
+
+    // branch with tip as commit to cherry pick
+    featureBranch = await getBranchOrError(repository, featureBranchName)
+
+    // branch with to cherry pick to
+    targetBranch = await getBranchOrError(repository, targetBranchName)
+
+    // set result to null for each test
+    result = null
+  })
+
+  it('successfully cherry pick one commit without conflicts', async () => {
+    result = await cherryPick(repository, featureBranch.tip.sha)
+    const cherryPickedCommit = await getCommit(
+      repository,
+      featureBranch.tip.sha
+    )
+
+    const commits = await getCommits(repository, targetBranch.ref, 3)
+    expect(commits.length).toBe(2)
+    expect(commits[0].summary).toBe(cherryPickedCommit!.summary)
+    expect(result).toBe(CherryPickResult.CompletedWithoutError)
+  })
+
+  it('successfully cherry pick a commit with empty message', async () => {
```

Comment:
Nope. I was also a little perplexed by that. (I actually still have it in my write up to the contrary) I originally had it grouped with the failures. But, I confirmed that this is adding a commit with an empty message and it does in fact cherry pick it. Thus, I am not sure why the documentation indicates that it would fail.

Reply:
I just added a couple more lines to the test to verify before cherry picking that the commit I am cherry picking does have an empty message. 🤷‍♀️

Reply:
Hmmm maybe the behavior changed in a more recent version of git 🤔
Anyway, that additional check you just added is 💯 , thank you!